pkg/daemon/agent/cluster/controller.go

```diff
@@ -67,38 +67,23 @@ func NewClusterController(context *clusterd.Context, flexvolumeController flexvo
 // StartWatch will start the watching of cluster events by this controller
 func (c *ClusterController) StartWatch(namespace string, stopCh chan struct{}) error {
 
-	customResourceClient, scheme, err := opkit.NewHTTPClient(k8sutil.CustomResourceGroup, k8sutil.V1Alpha1, opcluster.SchemeBuilder)
-	if err != nil {
-		return fmt.Errorf("failed to get a k8s client for watching cluster resources: %v", err)
-	}
-	c.scheme = scheme
-
 	resourceHandlerFuncs := cache.ResourceEventHandlerFuncs{
 		DeleteFunc: c.onDelete,
 	}
 
 	logger.Infof("start watching cluster resources")
-	watcher := opkit.NewWatcher(opcluster.ClusterResource, namespace, resourceHandlerFuncs, customResourceClient)
-	go watcher.Watch(&opcluster.Cluster{}, stopCh)
+	watcher := opkit.NewWatcher(opcluster.ClusterResource, namespace, resourceHandlerFuncs, c.context.RookClientset.Rook().RESTClient())
```

Comment:
Is there a reason we need the RestClient object? I thought this change would make it work more like the K8s resources.

Reply:
i did start to convert this to the new client, but did not find it to be as natural for CRDs. The add/update/delete methods are only accepted by the rest client. The new client would require us to implement a loop and respond to the different strongly typed events that wasn't as clean.